tap_aggregator/src/api_versioning.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023-, Semiotic AI, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use jsonrpsee::core::Serialize;
+use serde::Deserialize;
+use strum::{self, IntoEnumIterator};
+
+/// The versions of the TAP JSON-RPC API implemented by this server.
+/// The version numbers are independent of the TAP software version. As such, we are
+/// enabling the introduction of breaking changes to the TAP library interface without
+/// necessarily introducing breaking changes to the JSON-RPC API (or vice versa).
+#[derive(
+    Clone,
+    Debug,
+    Eq,
+    PartialEq,
+    strum::Display,
+    strum::EnumString,
+    strum::EnumVariantNames,
+    strum::EnumIter,
+)]
+pub enum TapRpcApiVersion {
+    #[strum(serialize = "0.0")]
+    V0_0,
+}
+
+// We implement our own Serialize and Deserialize traits for `TapRpcApiVersion` because
+// the ones derived by `serde` serialize the enum member names as strings (eg. "V0_0"),
+// while we want to serialize them using the variant strings we set through `strum`
+// (eg. "0.0").
+
+impl serde::ser::Serialize for TapRpcApiVersion {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for TapRpcApiVersion {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<TapRpcApiVersion, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        TapRpcApiVersion::from_str(&s).map_err(serde::de::Error::custom)
+    }
+}
+
+/// List of RPC version numbers for which a deprecation warning has to be issued.
+/// This is a very basic approach to deprecation warnings. The most important thing
+/// is to have *some* process in place to warn users of breaking changes.
+/// NOTE: Make sure to test it when that list becomes non-empty.
```

Comment:
This is not a change needed for approval. Just a thought on how we might test this now instead of waiting until list is non-empty. I'll fold it into an issue and we can look into it later. As we previously discussed you already did a manual test (filling in the array and testing it) so we should have some confidence.